### PR TITLE
Purge data from a different sync group whenever all syncs complete

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/1413/regularly-purge-data-outside-the-current-sync-group

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/sync/DataSync.kt [hunk 1]
```diff
@@ -48,7 +48,7 @@ class DataSync(
 
   @Inject
   constructor(
-      modelSyncs: ArrayList<ModelSync>,
+      modelSyncs: List<@JvmSuppressWildcards ModelSync>,
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/sync/DataSync.kt:52]: When do we need `JvmSuppressWildcards` ?

File: app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt [hunk 2]
```diff
@@ -687,6 +691,151 @@ class DataSyncTest {
...
+    )
+
+    val modelSync3 = FakeModelSync(
+        _name = "sync3",
+        config = dailySyncConfig,
+        pushError = RuntimeException()
+    )
+
+    val purgeOnSync = mock<PurgeOnSync>()
+    val dataSync = DataSync(
```
Comment (Honey14) [inline app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt:864]: Can we make `modelSync1` , `modelSync2` , `modelSync3`, `purgeOnSync` , `dataSync` as global variables? We're using them in most of the tests?

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks great. Left one question and a comment. PTAL

File: app/src/main/java/org/simple/clinic/sync/DataSync.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/sync/DataSync.kt:52]: Dagger has an issue with injecting wildcard generics (If you look at `SyncModule`, it provides `frequentlySyncingRepositories` as `List<SynceableRepositories<*, *>>` which are wildcards. This is a problem only when you use Kotlin, which has this annotation that you can use to fix this problem.

File: app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt:864]: The `modelSync` instances are not the same in the tests. They have the same names, but they behave differently.

**Review by Honey14 — APPROVED** [review]
Looks good

**Pull request merged**